### PR TITLE
Determine target branch based off current branch

### PR DESCRIPTION
**Pull request by kevincarrogan**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -111,10 +111,17 @@ commands:
...
-        default: master
+        default: auto
         type: string
     steps:
-      - run: git clone git@github.com:uktrade/lite-api.git; cd lite-api; git checkout <<parameters.api_branch>>
+      - run:
+          name: Clone API
+          command: |
+            echo "Checking out"
+            python which_branch.py <<parameters.api_branch>>
```
Comment (currycoder) [inline .circleci/config.yml:121]: Is this needed anymore?

File: which_branch.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return os.environ.get("CIRCLE_BRANCH", "dev")
+
+
+BASE_TARGET_MAP = {
+    "dev": "uat",
+    "uat": "master",
+}
+
+
+HOTFIX_BRANCH_RE = re.compile("^hotfix-(.*)-.*$")
```
Comment (currycoder) [inline which_branch.py:16]: So we need to agree in our hotfix process to prefix branch names with "hotfix-" right?

Comment (currycoder) [reply which_branch.py:16]: Ah actually "hotfix-{uat|master}"..?

Comment (currycoder) [reply which_branch.py:16]: I like this as the complexity is around the exceptional case rather than the regular one

**Review by Harriethw — APPROVED** [review]

**Review by currycoder — APPROVED** [review]

**Pull request merged**